# Explanatory Data Analysis and Seaborn Visualization - Olympic Games 

__Method__:

- Import the data / first inspection 

- Merging and concatenating 

- Data cleaning 


In [1]:
#Import Libraries 
import pandas as pd 

In [9]:
#Read in data for Olympic Games / Dictionary 
summer = pd.read_csv("summer.csv")
winter = pd.read_csv("winter.csv")
dc = pd.read_csv("dictionary.csv")

In [3]:
#Explore data / summer
summer

Year    City      Sport           Discipline  \
0      1896  Athens   Aquatics             Swimming   
1      1896  Athens   Aquatics             Swimming   
2      1896  Athens   Aquatics             Swimming   
3      1896  Athens   Aquatics             Swimming   
4      1896  Athens   Aquatics             Swimming   
...     ...     ...        ...                  ...   
31160  2012  London  Wrestling  Wrestling Freestyle   
31161  2012  London  Wrestling  Wrestling Freestyle   
31162  2012  London  Wrestling  Wrestling Freestyle   
31163  2012  London  Wrestling  Wrestling Freestyle   
31164  2012  London  Wrestling  Wrestling Freestyle   

                         Athlete Country Gender                       Event  \
0                  HAJOS, Alfred     HUN    Men              100M Freestyle   
1               HERSCHMANN, Otto     AUT    Men              100M Freestyle   
2              DRIVAS, Dimitrios     GRE    Men  100M Freestyle For Sailors   
3             MALOKINIS, Ioannis     GRE    Men  100M Freestyle For Sailors   
4             CHASAPIS, Spiridon     GRE    Men  100M Freestyle For Sailors   
...                          ...     ...    ...                         ...   
31160         JANIKOWSKI, Damian     POL    Men                    Wg 84 KG   
31161  REZAEI, Ghasem Gholamreza     IRI    Men                    Wg 96 KG   
31162             TOTROV, Rustam     RUS    Men                    Wg 96 KG   
31163          ALEKSANYAN, Artur     ARM    Men                    Wg 96 KG   
31164             LIDBERG, Jimmy     SWE    Men                    Wg 96 KG   

        Medal  
0        Gold  
1      Silver  
2      Bronze  
3        Gold  
4      Silver  
...       ...  
31160  Bronze  
31161    Gold  
31162  Silver  
31163  Bronze  
31164  Bronze  

[31165 rows x 9 columns]

 - We have data ranging from 1896 to 2012. 
 - Each and every row is one athlete who has won a medal for a given sport for the Summer games.
 - 31165 entries 

In [12]:
summer.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31165 entries, 0 to 31164
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Year        31165 non-null  int64 
 1   City        31165 non-null  object
 2   Sport       31165 non-null  object
 3   Discipline  31165 non-null  object
 4   Athlete     31165 non-null  object
 5   Country     31161 non-null  object
 6   Gender      31165 non-null  object
 7   Event       31165 non-null  object
 8   Medal       31165 non-null  object
dtypes: int64(1), object(8)
memory usage: 2.1+ MB


In [18]:
summer[summer.isnull().any(axis = 1)]

Year    City          Sport           Discipline          Athlete  \
29603  2012  London      Athletics            Athletics          Pending   
31072  2012  London  Weightlifting        Weightlifting          Pending   
31091  2012  London  Weightlifting        Weightlifting          Pending   
31110  2012  London      Wrestling  Wrestling Freestyle  KUDUKHOV, Besik   

      Country Gender     Event   Medal  
29603     NaN  Women     1500M    Gold  
31072     NaN  Women      63KG    Gold  
31091     NaN    Men      94KG  Silver  
31110     NaN    Men  Wf 60 KG  Silver

- 9 correctors and 6 attributes 
- Country columns has 4 missing values 
- Data types are correct for given values

In [7]:
#Explore data / winter
winter

Year      City     Sport Discipline               Athlete Country  \
0     1924  Chamonix  Biathlon   Biathlon           BERTHET, G.     FRA   
1     1924  Chamonix  Biathlon   Biathlon        MANDRILLON, C.     FRA   
2     1924  Chamonix  Biathlon   Biathlon   MANDRILLON, Maurice     FRA   
3     1924  Chamonix  Biathlon   Biathlon       VANDELLE, André     FRA   
4     1924  Chamonix  Biathlon   Biathlon  AUFDENBLATTEN, Adolf     SUI   
...    ...       ...       ...        ...                   ...     ...   
5765  2014     Sochi    Skiing  Snowboard          JONES, Jenny     GBR   
5766  2014     Sochi    Skiing  Snowboard       ANDERSON, Jamie     USA   
5767  2014     Sochi    Skiing  Snowboard    MALTAIS, Dominique     CAN   
5768  2014     Sochi    Skiing  Snowboard          SAMKOVA, Eva     CZE   
5769  2014     Sochi    Skiing  Snowboard      TRESPEUCH, Chloe     FRA   

     Gender            Event   Medal  
0       Men  Military Patrol  Bronze  
1       Men  Military Patrol  Bronze  
2       Men  Military Patrol  Bronze  
3       Men  Military Patrol  Bronze  
4       Men  Military Patrol    Gold  
...     ...              ...     ...  
5765  Women       Slopestyle  Bronze  
5766  Women       Slopestyle    Gold  
5767  Women  Snowboard Cross  Silver  
5768  Women  Snowboard Cross    Gold  
5769  Women  Snowboard Cross  Bronze  

[5770 rows x 9 columns]

 - From 1924 to 2014 
 - 5770 entries 
 - Each and every row represents a athlete who has won a medal for an Olympic winter game 

In [8]:
winter.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5770 entries, 0 to 5769
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Year        5770 non-null   int64 
 1   City        5770 non-null   object
 2   Sport       5770 non-null   object
 3   Discipline  5770 non-null   object
 4   Athlete     5770 non-null   object
 5   Country     5770 non-null   object
 6   Gender      5770 non-null   object
 7   Event       5770 non-null   object
 8   Medal       5770 non-null   object
dtypes: int64(1), object(8)
memory usage: 405.8+ KB


 - No missing values 
 - Data types are appropriate for given entries 

In [14]:
#Explore data / dictionary
dc

Country Code  Population  GDP per Capita
0        Afghanistan  AFG  32526562.0      594.323081
1            Albania  ALB   2889167.0     3945.217582
2            Algeria  ALG  39666519.0     4206.031232
3    American Samoa*  ASA     55538.0             NaN
4            Andorra  AND     70473.0             NaN
..               ...  ...         ...             ...
196          Vietnam  VIE  91703800.0     2111.138024
197  Virgin Islands*  ISV    103574.0             NaN
198            Yemen  YEM  26832215.0     1406.291651
199           Zambia  ZAM  16211767.0     1304.879014
200         Zimbabwe  ZIM  15602751.0      924.143819

[201 rows x 4 columns]

In [15]:
dc.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 201 entries, 0 to 200
Data columns (total 4 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country         201 non-null    object 
 1   Code            201 non-null    object 
 2   Population      196 non-null    float64
 3   GDP per Capita  176 non-null    float64
dtypes: float64(2), object(2)
memory usage: 6.4+ KB


 - 201 entries 
 - Missing values for Population / GDP per Capita 
 - Data types are correct for given values 

In [17]:
#Filter countries with missing values 
dc[dc.isnull().any(axis = 1)]

Country Code  Population  GDP per Capita
3                    American Samoa*  ASA     55538.0             NaN
4                            Andorra  AND     70473.0             NaN
9                             Aruba*  ARU    103889.0             NaN
20                          Bermuda*  BER     65235.0             NaN
27            British Virgin Islands  IVB     30117.0             NaN
36                   Cayman Islands*  CAY     59967.0             NaN
45                      Cook Islands  COK         NaN             NaN
49                              Cuba  CUB  11389562.0             NaN
61                           Eritrea  ERI         NaN             NaN
74                              Guam  GUM    169885.0             NaN
86                              Iran  IRI  79109272.0             NaN
96                      Korea, North  PRK  25155317.0             NaN
105                            Libya  LBA   6278438.0             NaN
106                    Liechtenstein  LIE     37531.0             NaN
116                       Mauritania  MTN   4067564.0             NaN
121                           Monaco  MON     37731.0             NaN
130            Netherlands Antilles*  AHO         NaN             NaN
139  Palestine, Occupied Territories  PLE         NaN             NaN
141                 Papua New Guinea  PNG   7619321.0             NaN
147                     Puerto Rico*  PUR   3474182.0             NaN
156                       San Marino  SMR     31781.0             NaN
176                            Syria  SYR  18502413.0             NaN
177                           Taiwan  TPE         NaN             NaN
195                        Venezuela  VEN  31108083.0             NaN
197                  Virgin Islands*  ISV    103574.0             NaN

 - Smaller countries/ Special cases of governments where information is not given.

# Merging and Concatenating Summer and Winter Data

__Method__: 
- Vertical concatenate
- Left join 

Concatenating points:
- Summer and winter data have the same 9 columns whoch makes it easy to concatenate.


